hdr_html_canvas_element.md

```diff
@@ -0,0 +1,744 @@
+Display light has the property that it is limited by a maximum and minimum value that the display device can produce.
+
+Display light is a precise physical quantity, namely, luminance in nits (nits are cd/m^2).
+
+Display light is sometimes treated as being in the interval [0, 1], with 1 representing the maximum luminance the device can produce, and 0 possibly representing the minimum luminance that the device can produce.
+In these cases display light is sometimes referred to as being "linear", because nothing more than a linear (or more precisely, affine) transformation has been applied to it.
+We will not use this convention.
+
+#### Scene light
+
+Scene light is light that is captured by a capture device (e.g, a camera) at each pixel.
+Scene light also has a maximum value, either imposed by the limitations of the capture device, or the known maximum value of the particular scene being captured.
+
+The precise units of scene light are vague.
+Under some circumstances, the units are precise number of nits of each input pixel.
+Under some circumstances, the units are normalized to the interval [0, 1].
+We will use the latter convention, with scene light being normalized such that the value of 1 represents diffuse white, and we will draw attention to situations where we do not follow this rule.
+
+Similarly to display light, scene light is sometimes referred to as being "linear".
+
+#### Signal
+
+A signal is a digial encoding of scene light.
+
+An example of a signal is the pixel values in an sRGB image.
+
+Most signals take values in the [0, 1] interval and are represented using fixed-point (e.g, the usual 8-bit encoding of an sRGB image).
+In this document, we will be allowing for signals that take any real value, and are represented using floating-point.
+
+### Transfer functions
+
+#### Opto-electronic transfer function (OETF)
+
+An opto-electronic transfer function (OETF) is a function that transforms scene light to a signal.
+
+An example of an OETF is the [sRGB function](https://www.w3.org/Graphics/Color/srgb) ``f(x)=(x<0.0031308) ? 12.92*x : 1.055*pow(x,1/2.4)-0.055``.
+The domain of this OETF (scene light) is the interval [0, 1], with diffuse white at 1 (which is what we adopted as convention).
+The range (signal) is standard sRGB pixel values in the interval [0, 1].
+
+Another example of an OETF is the HLG OETF, specified in Table 5 of the [BT.2100 specification](https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.2100-2-201807-I!!PDF-E.pdf).
+For this OETF, the domain (scene light) is normalized to the interval [0, 1], with maximum brightness at 1 (which we will treat as matching our convention).
+
+Another example of an OETF is the PQ OETF, specified in Table 4 of the [BT.2100 specification](https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.2100-2-201807-I!!PDF-E.pdf).
+For this OETF, the domain (scene light) is luminance values in nits (which does not not match our convention, and we will discuss how to match this in detail).
```

Comment:
Is this correct? In table 4, the OETF maps relative scene linear light into the non-linear PQ signal value.

Reply:
This is a good example of the definition problem that I was discussing. (Short answer: "you're right").

What is "relative scene linear light"? BT2100 uses it, but gives no definition. From its use, and with some detective work, we can piece together what might be a definition that is consistent with its usage.

In Table 4 that "Input signal to PQ opto-optical transfer function (OOTF)" is "Scene linear light. The OOTF maps relative scene linear light to display linear light." Well, which is it, "scene linear light" or "relative scene linear light"? They're different words, so presumably they have different meanings? What does "relative" mean here? Why is the word "linear" in there?

The next row in Table 4 contains the symbols "OOTF[E]", so I guess this means that E is the input signal to the OOTF which we established is either "scene linear light" or "relative scene linear light". (let's just go with "relative scene linear light"). And then E is defined as "E is the signal determined by scene light and scaled by camera exposure". No definition for "scene light" either, but at least that one is intuitive.

Table 5 has the same text, except defines E as "E is a signal for each colour component ... proportional to scene linear light normalized to the range [0:1]".

If we go back to Table 4, we can see that in the "Reference PQ OOTF" row, "E′ = G_{709}[E]" is defined only for values of E in the half-open interval [0, 1). So that supports the hypothesis that relative scene linear light is limited to the unit interval (perhaps closed or perhaps half-open).

Putting all of this together, we could say that "Relative scene linear light (denoted by E in this document), is a value in the interval [0, 1] that is linearly proportional to the physically measurable light in the scene being captured. The precise scaling factor used in mapping from physically measurable light to relative scene linear light depends on the particular capture setup, including such variables as camera exposure".

This is the sort of definition that I want to put at the top of the proposal -- I want it to be readable to people who have a basic understanding of compute graphics and the physics of light, but who are not steeped in broadcast terminology.